device/isis/segmentrouting.yaml

```diff
@@ -0,0 +1,215 @@
+components:
+  schemas:
+    Isis.SegmentRouting:
+      description: |-
+        Segment Routing (SR) turns on any node to select any path (explicit or derived from IGPs Shortest Path First (SPT) computations) 
+        for each of its traffic classes. The path does not depend on a hop-by-hop 
+        signaling technique (unlike LDP or RSVP). 
+        It only depends on a set of segments that are advertised by the IS-IS routing protocol. 
+        These segments act as topological sub-paths that can be combined together to form the required path.
+        Reference: https://datatracker.ietf.org/doc/html/rfc8667
+      type: object
+      properties:
+        router_capability:
+          description: |-
+            Optional IS-IS TLV named CAPABILITY, formed of multiple sub-TLVs, which allows a router to announce its
+            capabilities within an IS-IS level or the entire routing domain.        
+          $ref: '#/components/schemas/Isis.RouterCapability'
+          x-field-uid: 1
+      
+    Isis.RouterCapability:
+      description: >-
+        Container for the configuration of IS-IS Router CAPABILITY TLV.
+        https://datatracker.ietf.org/doc/html/rfc7981#section-2
+      type: object
+      properties:
+        choice:
+          description: |-
+            The Router Capability ID SHOULD be identical to the value advertised in the Traffic Engineering Router ID TLV [RFC5305].
+            If no Traffic Engineering Router ID is assigned, the Router ID SHOULD be identical to an IP Interface Address [RFC1195] 
+            advertised by the originating IS.
+            If the originating node does not support IPv4, then the reserved value 0.0.0.0 MUST be used in the Router ID field, 
+            and the IPv6 TE Router ID sub-TLV [RFC5316] MUST be present in the TLV.
+            - ipv4_te_router_id: IPv4 Traffic Engineering(TE) router id (defined in isis.basic.ipv4_te_router_id) to be used as Router Capability ID.
+            - interface_ip: When this is chosen the first IPv4 address of the first eth interface associated with this isis router instance should be assigned as the Router Capability ID.
+            - custom_router_cap_id: This option should be chosen when Router Capability ID needs to be configured different from above two options.
+          type: string
+          default: ipv4_te_router_id
+          x-field-uid: 1
+          x-enum:
+            ipv4_te_router_id:
+              x-field-uid: 1
+            interface_ip:
+              x-field-uid: 2
+            custom_router_cap_id:
+              x-field-uid: 3
+        custom_router_cap_id:
+          description: >-
```

Comment:
Maybe good to set a default of 0.0.0.0 just in case choice is directly set without setting the field itself to avoid language specific behaviour of the value sent.